spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/batch/AbstractSimpleJobServiceTests.java

```diff
@@ -192,12 +194,24 @@ void exceptionsShouldBeThrownIfRequestForNonExistingJobInstance() {
 	@Test
 	void stoppingJobExecutionShouldLeaveJobExecutionWithStatusOfStopping() throws Exception {
 		JobExecution jobExecution = createJobExecution(BASE_JOB_INST_NAME,true);
+		assertJobHasStopped(jobExecution);
+	}
+
+	@Test
+	void stoppingAllJobExecutionsShouldLeaveJobExecutionsWithStatusOfStopping() throws Exception {
+		JobExecution jobExecutionOne = createJobExecution(BASE_JOB_INST_NAME,true);
+		JobExecution jobExecutionTwo = createJobExecution(BASE_JOB_INST_NAME+"_TWO",true);
+		assertJobHasStopped(jobExecutionOne);
+		assertJobHasStopped(jobExecutionTwo);
+	}
+
+	private void assertJobHasStopped(JobExecution jobExecution) throws NoSuchJobExecutionException, JobExecutionNotRunningException {
 		jobExecution = jobService.getJobExecution(jobExecution.getId());
 		assertThat(jobExecution.isRunning()).isTrue();
 		assertThat(jobExecution.getStatus()).isNotEqualTo(BatchStatus.STOPPING);
 		jobService.stop(jobExecution.getId());
 		jobExecution = jobService.getJobExecution(jobExecution.getId());
-		assertThat(jobExecution.getStatus()).isEqualTo(BatchStatus.STOPPED);
+		assertThat(jobExecution.getStatus()).isEqualTo(BatchStatus.STOPPING);
```

Comment:
should we wait for STOPPED as well?

Reply:
`STOPPED` only occurs once the batch job has completed.  In this test we are just testing that SCDF sets the state to `STOPPING`.   We are not running real jobs here, just creating / modifying the state of the `JobExecution`.